Sources/VGSCollectSDK/UIElements/Text Field/VGSExpDateTextField.swift

```diff
@@ -132,10 +123,35 @@ extension VGSExpDateTextField: UIPickerViewDelegate, UIPickerViewDataSource {
       if pickerView.selectedRow(inComponent: yearPickerComponent) == 0 && pickerView.selectedRow(inComponent: monthPickerComponent) < currentMonthIndex {
         pickerView.selectRow(currentMonthIndex, inComponent: monthPickerComponent, animated: true)
       }
-      updateTextField()
+      updateTextFieldWithDatePickerSelection()
     }
 }
 
+// MARK: -  Configuration
+
+private extension VGSExpDateTextField {
+  
+  /// setup date picker configuration
+  private func setupDatePicker() {
+    textField.inputView = picker
+    updateYearsDataSource()
+    updateMonthsDataSource()
+    scrollToCurrentMonth(animated: false)
+    textField.inputAccessoryView = UIView()
+  }
+
+  /// setup keyboard configuration
+  private func setupKeyboard(with configuration: VGSExpDateConfiguration) {
+    textField.keyboardType = configuration.keyboardType ?? configuration.type.keyboardType
```

Comment:
Would suggest to hardcode number keyboard.

Reply:
default keyboard for `.expDate` is decimal. We give some flexibility to choose other keyboard if needed. Format pattern will not allow to type other then digits symbols.